target/linux/x86/64/config-5.4

```diff
@@ -159,6 +159,7 @@ CONFIG_EFI_RUNTIME_WRAPPERS=y
 CONFIG_EFI_STUB=y
 # CONFIG_EFI_TEST is not set
 # CONFIG_EFI_VARS is not set
+CONFIG_ENA_ETHERNET=y
```

Comment:
IMHO it is better to build this as a module and not include it into the default kernel.

Reply:
Since these features are needed to access the root filesystem and use default network, enable them by default in x86_64 kernel.

Reply:
if I build this as a module I would still have to add it to the default packages for x86-64 target, because as mentioned this is the default (and only) type of network interface of these cloud instances.
Would that be significantly different than just enabling it in the kernel?
In x86-64 kernel config there are a lot of various virtual/paravirtual drivers already for kvm/xen, should this be different?

Reply:
> Would that be significantly different than just enabling it in the kernel?

Yes, you can remove the module using the imagebuilder then.

Reply:
Indeed, good point!

Reply:
Note that I was only referring to the network adapter, which most are already (although it being mostly physical adapters) distributed as modules.

Yes, this was done very inconsequent in the past for virtualization targets (x86 is not the most well maintained in that regard), however i wouldn't let this hold us back to opt for the more modual approach.

> Ideological dislike for Amazon perhaps?

My credit card statements do not back this assumption.

Reply:
(sigh) fine, you win, I'll make a module.

Since I'm going to edit makefiles anyway, can I also move all network devices of other virtualization targets as modules (and add the modules as default) so that the situation is now consistent?

I can test only KVM and Vmware though, although just moving to kmods shouldn't affect functionality.